test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/MiddlewareFilterConfigurationProviderTest.cs

```diff
@@ -14,6 +14,19 @@ namespace Microsoft.AspNetCore.Mvc.Internal
     public class MiddlewareFilterConfigurationProviderTest
     {
         [Fact]
+        public void CreateConfigureDelegate_ThrowsIfNoConstructorFoundForType()
```

Comment:
Also try specifying a scenario with an interface. You could make this test use a `Theory` to test these scenarios.